src/bp/lang-server/api.ts

```diff
@@ -175,69 +177,72 @@ export default async function(options: APIOptions) {
         throw new BadRequestError('Param `tokens` is mandatory (must be an array of strings)')
       }
 
-      const result = await options.service.vectorize(tokens, lang)
+      const result = await options.languageService.vectorize(tokens, lang)
       res.json({ language: lang, vectors: result })
     } catch (err) {
       next(err)
     }
   })
 
   const router = express.Router({ mergeParams: true })
-
-  router.get('/list', (req, res, next) => {
-    const items = options.manager.inProgress.map(x => ({
-      status: x.getStatus(),
+  router.get('/', (req, res, next) => {
+    const downloading = options.downloadManager.inProgress.map(x => ({
       lang: x.lang,
-      id: x.id,
-      downloadedSize: x.downloadedSize,
-      fileSize: x.fileSize
+      progress: {
+        status: x.getStatus(),
+        downloadId: x.id,
+        size: x.downloadSizeProgress
+      }
     }))
 
     res.send({
-      available: options.manager.available,
-      installed: options.service.listFastTextModels().map(x => ({
-        lang: x.name,
-        loaded: x.loaded,
-        dim: options.service.dim,
-        domain: options.service.domain
-      })),
-      in_progress: items
+      available: options.downloadManager.downloadableLanguages,
+      installed: options.languageService.getModels(),
+      downloading
     })
   })
 
-  router.post('/install/:lang', (req, res, next) => {
+  router.post('/:lang', DisabledReadonlyMiddleware(options.readOnly), async (req, res, next) => {
     const { lang } = req.params
     try {
-      options.manager.download(lang)
-      res.sendStatus(200)
+      const downloadId = await options.downloadManager.download(lang)
+      res.status(200).send({ success: true, downloadId })
     } catch (err) {
       res.status(404).send({ success: false, error: err.message })
     }
   })
 
-  router.post('/remove/:lang', (req, res, next) => {
+  router.delete('/:lang', DisabledReadonlyMiddleware(options.readOnly), async (req, res, next) => {
```

Comment:
You can create an instance of the MW instead of passing options:
this.disabledMw = disabledReadonlyMw(options.readonly)